libminifi/src/io/tls/TLSSocket.cpp

```diff
@@ -31,14 +32,11 @@ namespace nifi {
 namespace minifi {
 namespace io {
 
-std::atomic<TLSContext*> TLSContext::context_instance;
-std::mutex TLSContext::context_mutex;
-
-TLSContext::TLSContext()
-    : error_value(0),
+TLSContext::TLSContext(std::shared_ptr<Configure> configure)
+    : SocketContext(configure), error_value(0),
       ctx(0),
       logger_(logging::Logger::getLogger()),
-      configuration(Configure::getConfigure()) {
+      configure_(configure) {
```

Comment:
for some reason configure_ seems wrong here. sounds petty but Configure as the name of the Configuration class seems odd to me.